src/actions.js

```diff
@@ -73,14 +78,48 @@ export function callWaiter(name, { requestCreator, params }) {
       );
     }
 
-    const request = waiterData.requestCreator(params, dispatch);
-    dispatch(initRequest(name, { request, params }));
+    const request = waiterData.requestCreator(waiterData.params, dispatch);
+    dispatch(initRequest(name, { request }));
     request
       .then((data) => {
+        const preEndState = getWaiter(getState(), name);
+
+        // waiter was destroyed
+        if (preEndState.name === null) {
+          return null;
+        }
+
+        // A new request was made, should not listen to this request anymore
+        if (preEndState.id !== waiterData.id) {
+          return null;
+        }
+
+        // Request was canceled
+        if (preEndState.isCanceled) {
+          return null;
+        }
+
```

Comment:
This is what I was working on, and wasn't sure if isCanceled was needed.

Reply:
This currently doesn't past tests, and is just in a WIP state. If the waiter is destroyed, it probably doesn't need a check. If the waiter id changes, that could have a strange effect on the model if an earlier request overwrites a later request. Still musing about on it